proxy/src/inbound.rs

```diff
@@ -74,7 +74,9 @@ where
         let &(ref addr, ref proto) = key;
         debug!("building inbound {:?} client to {}", proto, addr);
 
-        Buffer::new(self.bind.bind_service(addr, proto), self.bind.executor())
+        let endpoint = (*addr).into();
+        let bind = self.bind.bind_service(&endpoint, proto);
```

Comment:
TIOLI: seems like `bind_service` _could_ be changed to take `I: Into<Self::Endpoint>`? Though I'm not sure if that will continue to be useful when `Endpoint` owns metadata...